server/game/core/gameSteps/abilityWindow/TriggeredAbilityWindow.ts

```diff
@@ -45,31 +44,26 @@ export class TriggeredAbilityWindow extends BaseStep {
     ) {
         super(game);
 
-        this.toStringName = `'TriggeredAbilityWindow: ${this.eventWindow.events.map((event) => event.name).join(', ')}'`;
+        this.triggeringEvents = [...this.eventWindow.events];
     }
 
-    public emitEvents() {
-        this.eventsEmitted = true;
+    public addTriggeringEvents(events:GameEvent[]) {
+        this.triggeringEvents.push(...events);
+    }
 
-        const events = this.eventWindow.events.filter((event) => !this.eventsToExclude.includes(event));
+    public emitEvents() {
+        const events = this.triggeringEvents.filter((event) => !this.eventsToExclude.includes(event));
```

Comment:
I missed this previously - can we add a check here that also filters out any events that are already cancelled before we emit them

Reply:
Done.